shopify/shopify_custom_pixel/cl_custom_pixel_code.js

```diff
@@ -0,0 +1,410 @@
+                case "productType":
+                    CLabs_product.product_type = {"t": "string", "v": products[i].productType}
+                break;
+                case "coupon":
+                    CLabs_product.product_coupon = {"t": "string", "v": products[i].coupon}
+                break;
+                case "quantity":
+                    CLabs_product.product_quantity = {"t": "string", "v": products[i].quantity}
+                break;
+                default:
+                    let isnum = /^\d+$/.test(products[i][key]);
+                    var type = ""
+                    if(Array.isArray(products[i][key])) {
+                        continue;
+                    } else if(typeof products[i][key] === 'object' && products[i][key] !== null && !Array.isArray(products[i][key])) {
+                        continue;
+                    } else if(products[i][key] !== null && (Number.isFinite(products[i][key]) || isnum || !isNaN(parseFloat(products[i][key])))) {
+                        type = "number"
+                    }  else {
+                        type = "string"
+                    }
+                    if(key.substring(0, 8) === "product_") {
+                        CLabs_product[key] = {"t": type, "v": products[i][key]}
+                    } else {
+                        CLabs_product["product_"+key] = {"t": type, "v": products[i][key]}
+                    }
+            }
+        }
+        CLabs_product_data.push(CLabs_product);
+    }
+    return CLabs_product_data
+}
+/**
+ * @function shopify_products_mapping
+ * @params items array
+ * function Converts product list to customerlabs product properties format
+**/
+function shopify_products_mapping(items) {
+   var products_list = [];
+   var products_length = items.length;
+   if(products_length > 0){
+     for(var i=0;i< products_length;i++){
+       var products = {
+          'id'              : items[i].variant.product.id,
+          'sku'             : items[i].variant.sku,
+          'name'            : items[i].variant.product.title,
+          'price'           : items[i].variant.price.amount,
+          'imageURL'        : items[i].variant.image.src, 
+          'brand'           : items[i].variant.product.vendor,
+          'quantity'        : items[i].quantity
+        }
+        products_list.push(products)
+     }
+   }
+   return products_list;
+}
+//pageview event
+analytics.subscribe("page_viewed", event => {
+    var properties = {
+        "customProperties": {
+            "page_url":{
+                "t": "string",
+                "v": event.context.window.location.href
+            }
+        }
+    };
+    _cl_page_viewed = setInterval(function(){
+        if(window.CLabsgbVar && window.CLabsgbVar.generalProps){
+            _cl.pageview('pageview',properties);
+            clearInterval(_cl_page_viewed)
+        }
+    }, 1000);
+});
+//Product viewed event
+analytics.subscribe("product_viewed", event => {
```

Comment:
@thisissathish    refer these things from the old code4 and update 
https://github.com/customerlabs/customerlabs_js_helpers/blob/70fe9d03c265663ef6c13af82397ea69fc2b9b5c/shopify/cl_default_ecommerce_events.liquid#L379